File: dag1/makeThread.py
from threading import Thread

i = 0

def someThreadFunction1():
	global i	
	while i<1000000:
		i+=1
	print ("first\n")

def someThreadFunction2():
	global i	
	while i<1000000:
		i+=1
	print ("second\n")


def main():
	someThread1 = Thread(target = someThreadFunction1, args = (),)
	someThread1.start()
	someThread2 = Thread(target = someThreadFunction2, args = (),)
	someThread2.start()

	someThread1.join()
	someThread2.join()
	print("ferdig!\n")

main()

File: dag1/makeThread.c
//gcc -std=gnu99 -Wall -g -o makeThread makeThread.c -lpthread

#include <pthread.h>
#include <stdio.h>

int i = 0;
void* someThreadFunction1(){
	for (int j=0; j <10000000; j++){
		i++;
	}
	return NULL;
}

void* someThreadFunction2(){
	for (int j=0; j <10000000; j++){
		i--;
	}
	return NULL;
}

int main(){
	pthread_t someThread1;
	pthread_t someThread2;
	pthread_create(&someThread1, NULL, someThreadFunction1, NULL);
	// Arguments to a thread would be passed here ---------^
	pthread_create(&someThread2, NULL, someThreadFunction2, NULL);

	pthread_join(someThread1, NULL);
	pthread_join(someThread2, NULL);
	printf(i);
	return 0;
}
